alleycats-core/src/main/scala/alleycats/syntax/extract.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2015 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package alleycats
+package syntax
+
+import alleycats.Extract
+
+object extract extends ExtractSyntax
+
+trait ExtractSyntax {
+  implicit class ExtractOps[F[_], A](fa: F[A])(implicit ev: Extract[F]) {
+    def extract(): A = ev.extract(fa)
+  }
```

Comment:
I wonder if it makes sense to convert it into a value class, e.g.:
```suggestion
  implicit final class ExtractOps[F[_], A](private val fa: F[A]) extends AnyVal {
    def extract(implicit ev: Extract[F]): A = ev.extract(fa)
  }
```
The reason why I think it could make sense is that
1. This is a pretty valid use case for value classes in general and allows to save on memory allocations a little bit.
2. Newer syntaxes in Cats-core use this approach a lot, see `Applicative` syntax for example:
    <details>

    https://github.com/typelevel/cats/blob/769bf21b83a81a070868b878414a21012d78dd9d/core/src/main/scala/cats/syntax/applicative.scala#L37-L56

    </details>